client-js/app/elements/labrad-plot1d.ts

```diff
@@ -306,11 +307,17 @@ export class Plot1D extends polymer.Base implements Plot {
             x = Math.max(0, Math.min(this.width, x));
             y = Math.max(0, Math.min(this.height, y));
             if (x !== originX && y !== originY) {
-              this.zoom.x(this.xScale.domain([originX, x].map(this.xScale.invert).sort()))
-                  .y(this.yScale.domain([originY, y].map(this.yScale.invert).sort()));
+              var xScale = this.xScale,
+                  yScale = this.yScale,
+                  xMin = Math.min(xScale.invert(originX), xScale.invert(x)),
+                  xMax = Math.max(xScale.invert(originX), xScale.invert(x)),
+                  yMin = Math.min(yScale.invert(originY), yScale.invert(y)),
+                  yMax = Math.max(yScale.invert(originY), yScale.invert(y));
+              this.zoom.x(xScale.domain([xMin, xMax]))
+                       .y(yScale.domain([yMin, yMax]));
```

Comment:
Would it be good to have a comment somewhere around here about making sure we don't flip the axes?